README.md

```diff
@@ -44,3 +51,93 @@ the webhook to send all necessary event types to the endpoint where the bot
 is running. Two things to remember is that make sure you send the webhook events
 to the `/postreceive` endpoint off of the server's address and that the
 `Content type` is set to `application/json`.
+
+
+### Per repo configuration
+
+qiskit-bot gives projects some local configuration options that can be set in
+the repository. To set a local configuration file a file `qiskit_bot.yaml` must
+be created in the root of the git repository. If this file is present then
+qiskit-bot will read it before every action and adjust behavior based on it's
+contents. Currently this configuration file is used to control two things,
+the changelog generation behavior and whether the bot will leave comments
+on new pull requests when they're opened (and the exact behavior of that
+comment). n example of a fully populated configuration file is:
+
+```yaml
+---
+categories:
+    "Changelog: Custom": Special category
+    "Changelog: Custom 2": Less special category
+    "Nothing": null
+notifications:
+    ".*":
+        - "@core-team"
+    qiskit/transpiler:
+        - "@user1"
+        - "@user2"
+    qiskit/transpiler/passes:
+        - "@user3"
+        - "@user4"
+always_notify: true
+notification_prelude: |
+    This is a custom prelude
+
+    I include whitespace:
+
+```
+
+The details on each option are as follows:
+
+- `categories`: This contains a nested mapping of github labels to changelog
+  sections. If specified at release time when qiskit-bot generates the changelog
+  it will look at each merged PR in the release and if any have any matching
+  labels that commit summary message will be put under the corresponding
+  sections in the changelog used for the release page. If a value for any label
+  is set to `null` this means that this label is counted as matching but will
+  not be included in the generated changelog. By default any labels outside this
+  set will not be included in the changelog but when the
+  `tools/generate_changelog.py` script is run it flags any merged PRs for a pending
+  release that don't have a matching label. By setting one or more labels to `null`
+  these PRs will not show up in that script.
+
+  If this field is not specified the following default values are used:
+  ```yaml
+  "Changelog: Deprecation": Deprecated
+  "Changelog: New Feature": Added
+  "Changelog: API Change": Changed
+  "Changelog: Removal": Removed
+  "Changelog: Bugfix": Fixed
+  "Changelog: None": null
+  ```
+
+- `notifications`: This contains a mapping of path regexes to a list of usernames
+  to notify if an opened PR touches files that match a particular regex (as found
+  by Python's stdlib `re.search()` function). For example if you set the path regex
+  to `".*"` this would match everything, but using a regex gives you control over
```

Comment:
It might be better to use/mention single quotes for regex strings - my understanding is that single-quoted YAML strings don't parse escape characters, but double-quotes do, and literal `\` is needed in regexes fairly commonly.

Reply:
Yeah, that's a fair point I had forgot that detail from the yaml spec (https://yaml.org/spec/1.2.2/#double-quoted-style vs https://yaml.org/spec/1.2.2/#single-quoted-style for reference). I'll update this to mention that